### ISSUE TITLE
cargo stops executing jobs for some time

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/cargo#7954]
I was building the compiler in one terminal `x.py build` and in a different terminal, I was building clippy `cargo +master test`.

Looking at `htop`, I could see that the rustc build was using 4 threads to build rustc, but the clippy build was not consuming any cpu load and seemed to stall for several minutes.
It hardly progressed (started building new crate deps) during that time, only one or two times.
Only after the rustc build was done, the `cargo build` would start spawning new jobs again and I am wondering what was happening because it looked like one `cargo build` made another `cargo build` process stall.
Is this supposed to happen?

rustc repo @ beac68a88711a90346ec8b68e3baefbec62b3b0d
`cargo 1.43.0-nightly (e57bd0299 2020-02-21)`

Output of `cargo version`:

OS: linux
I'm using sccache.

**Comment by ehuss** [comment]
Can you reproduce this without sccache?

In general, if you are having problems, it might be best to try without sccache first to see if that is the problem.  This issue otherwise isn't actionable since it is so vague.  Independent Cargo processes have locks when updating the index, but otherwise shouldn't block one another (unless they are both spawned from `make` using a jobserver).

**Comment by weihanglo** [comment]
I am going to close due to lack of a minimal set of steps to reproduce, and it has been inactive for a while. If you have any update for reproducing this without sccache, free to comment below or open a new issue.
